pkg/plugin/plugin.go

```diff
@@ -525,6 +525,39 @@ func (p *plugin) ListClusterVMs(ctx context.Context, cs *api.OpenShiftManagedClu
 	return &api.GenevaActionListClusterVMs{VMs: &pods}, nil
 }
 
+func (p *plugin) Restart(ctx context.Context, cs *api.OpenShiftManagedCluster, hostname string) error {
+	if !validate.IsValidAgentPoolHostname(hostname) {
+		return fmt.Errorf("invalid hostname %q", hostname)
+	}
```

Comment:
rebase and include a call to `checkIfClusterWillRefresh` here like the other geneva actions now do

Reply:
added